datadog/table_datadog_service_level_objective.go

```diff
@@ -22,64 +27,118 @@ func tableDatadogServiceLevelObjective(ctx context.Context) *plugin.Table {
-			{Name: "thresholds", Type: proto.ColumnType_JSON, Description: "Thresholds that are set for the SLOs."},
+			{Name: "description", Type: proto.ColumnType_JSON, Description: "Description of the SLO.", Transform: transform.FromField("Attributes.Description")},
+			{Name: "groups", Type: proto.ColumnType_JSON, Description: "A list of (up to 20) monitor groups that narrow the scope of a monitor service level objective.", Transform: transform.FromField("Attributes.Groups")},
+			{Name: "monitor_ids", Type: proto.ColumnType_JSON, Description: "A list of monitor ids that defines the scope of a monitor service level objective.", Transform: transform.FromField("Attributes.MonitorIDs")},
+			{Name: "query", Type: proto.ColumnType_JSON, Description: "The Metric based SLOs use queries to determine the state. Shows associated query.", Transform: transform.FromField("Attributes.Query")},
+			{Name: "monitor_tags", Type: proto.ColumnType_JSON, Description: "If monitors that are associated with SLO have tags they will show here.", Transform: transform.FromField("Attributes.MonitorTags")},
+			{Name: "tags", Type: proto.ColumnType_JSON, Description: "Tags associated with SLO.", Transform: transform.FromField("Attributes.AllTags")},
+			{Name: "thresholds", Type: proto.ColumnType_JSON, Description: "Thresholds that are set for the SLOs.", Transform: transform.FromField("Attributes.Thresholds")},
 		},
 	}
 }
 
+// Using `apiClient.ServiceLevelObjectivesApi.SearchSLO` doesn't populate the response and returns empty rows.
+// Additionally, `apiClient.ServiceLevelObjectivesApi.ListSLOs(ctx, opts)` does not return SLOs of the "By Time Slice" type.
+// Therefore, we opted for a raw API call to list all the SLOs (as is done in the Datadog console). This approach also addresses the issue: https://github.com/turbot/steampipe-plugin-datadog/issues/63.
+
 func listSLOs(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
 	ctx, apiClient, err := connectV1(ctx, d)
 	if err != nil {
 		plugin.Logger(ctx).Error("datadog_service_level_objective.listSLOs", "connection_error", err)
 		return nil, err
 	}
 
-	opts := datadog.ListSLOsOptionalParameters{}
+	steampipeConfig := GetConfig(d.Connection)
+
+	keys := ctx.Value(datadog.ContextAPIKeys).(map[string]datadog.APIKey)
+	apiKey := keys["apiKeyAuth"].Key
+	appKey := keys["appKeyAuth"].Key
+
+	pageNumber := 0
+	for {
+
+		params := url.Values{}
+		params.Add("query", "")
+		params.Add("sort", "")
+		params.Add("include_facets", "false")
+		params.Add("include_permissions", "true")
+		params.Add("page[size]", "1")
+
+		params.Add("page[number]", fmt.Sprint(pageNumber))
+
+		fullUrl := fmt.Sprintf("%s?%s", *steampipeConfig.ApiURL+"api/v1/slo/search", params.Encode())
+
+		plugin.Logger(ctx).Error("Full URL : ", fullUrl)
+		buildReq, err := http.NewRequest("GET", fullUrl, nil)
+		if err != nil {
+			fmt.Println("Error creating the request:", err)
```

Comment:
Should we be using `plugin.Logger(ctx).Error()` here?

Reply:
Done!